python/pyarrow/_s3fs.pyx

```diff
@@ -236,16 +236,24 @@ cdef class S3FileSystem(FileSystem):
             S3FileSystem(proxy_options={'scheme': 'http', 'host': 'localhost',
                                         'port': 8020, 'username': 'username',
                                         'password': 'password'})
-    allow_bucket_creation : bool, default False
+    allow_bucket_creation : boolean, default False
```

Comment:
Above doc uses `boolean` instead of `bool`

Reply:
While above it indeed uses `boolean`, doing a quick grep for this pattern in the full code base, we use `bool` much more often, so I would stick with bool here

Reply:
Yes, I think we should switch to `bool` for all these docstrings, since that's the proper type name in Python.